packages/react-ui/src/lib/data-selector-utils.ts

```diff
@@ -0,0 +1,153 @@
+export type MentionTreeNode = {
+  key: string;
+  data: {
+    propertyPath: string;
+    /**Key for json value */
+    displayName: string;
+    /**value for json key */
+    value?: string | unknown;
+    isSlice: boolean;
+    isTestStepNode?: boolean;
+  };
+  children?: MentionTreeNode[];
+};
+/**Traverses an object to find its child properties and their paths, stepOutput has to be an object on first invocation */
+function traverseStepOutputAndReturnMentionTree({
+  stepOutput,
+  propertyPath,
+  displayName,
+}: {
+  stepOutput: unknown;
+  propertyPath: string;
+  displayName: string;
+}): MentionTreeNode {
+  if (stepOutput && typeof stepOutput === 'object') {
+    if (Array.isArray(stepOutput)) {
+      return handlingArrayStepOutput(stepOutput, propertyPath, displayName);
+    }
+    return {
+      key: propertyPath,
+      data: {
+        propertyPath: propertyPath,
+        displayName: displayName,
+        isSlice: false,
+        value: Object.keys(stepOutput).length === 0 ? 'Empty List' : undefined,
+      },
+      children: Object.keys(stepOutput).map((k) => {
+        const escapedKey = k
+          .replaceAll(/\\/g, '\\')
+          .replaceAll(/"/g, '\\"')
+          .replaceAll(/'/g, "\\'")
+          .replaceAll(/\n/g, '\\n')
+          .replaceAll(/\r/g, '\\r')
+          .replaceAll(/\t/g, '\\t')
+          .replaceAll(/’/g, '\\’');
+        const newPath = `${propertyPath}['${escapedKey}']`;
+        const newDisplayName = k;
```

Comment:
both of these can be in-lined, they are only used once.

Ie.


```suggestion
     return traverseStepOutputAndReturnMentionTree({
          stepOutput: value,
          propertyPath: `${propertyPath}['${escapedKey}']`,
          displayName: key,
        });
```